resource-managers/kubernetes/core/src/main/scala/org/apache/spark/scheduler/cluster/kubernetes/KubernetesClusterSchedulerBackend.scala

```diff
@@ -88,6 +94,28 @@ private[spark] class KubernetesClusterSchedulerBackend(
       throw new SparkException(s"Executor cannot find driver pod", throwable)
   }
 
+  private val shuffleServiceConfig: Option[ShuffleServiceConfig] =
+    if (Utils.isDynamicAllocationEnabled(sc.conf)) {
+      val shuffleNamespace = conf.get(KUBERNETES_SHUFFLE_NAMESPACE)
+      val parsedShuffleLabels = Util.parseKeyValuePairs(
+        conf.get(KUBERNETES_SHUFFLE_LABELS), KUBERNETES_SHUFFLE_LABELS.key,
+            "shuffle-labels")
+      if (parsedShuffleLabels.size == 0) {
+        throw new SparkException(s"Dynamic allocation enabled " +
+          s"but no ${KUBERNETES_SHUFFLE_LABELS.key} specified")
+      }
+
+      val shuffleDirs = conf.getOption(KUBERNETES_SHUFFLE_DIR.key).map {
```

Comment:
I was trying to get Option[String] because if the shuffle directory is left empty, we use the default from `Utils.getConfiguredLocalDirs(conf)`. I'm not sure how we can get this behavior using `get`.